lib/ts/ink_defs.h

```diff
@@ -130,6 +145,12 @@ max(const T a, const T b)
 
 #if TS_USE_HWLOC
 #include <hwloc.h>
+#if HWLOC_API_VERSION < 0x00010b00
+#define HWLOC_OBJ_NUMANODE  HWLOC_OBJ_NODE
+#endif
+#elif __cplusplus
+// some signatures need this
+using hwloc_obj_type_t = int; 
```

Comment:
Typedef is not deprcated .. but it's much harder to read and it cannot easily work the the "default" situations in a stack of derived classes.  It's nice to just "using SubClass::TypeInSubclass_t;" and then ... frankly .. *all* types should be introduced that way.